compiler/util/llvmDebug.cpp

```diff
@@ -0,0 +1,743 @@
+  int line_number = varSym->astloc.lineno;
+  FnSymbol *funcSym = NULL;
+  if(isFnSymbol(varSym->defPoint->parentSymbol))
+    funcSym = (FnSymbol*)varSym->defPoint->parentSymbol;//TODO:if parent is a block
+  else 
+    printf("Couldn't find the function parent of variable: %s!\n",name);
+  
+  LLVM_DISUBPROGRAM scope = get_function(funcSym);  
+  LLVM_DIFILE file = get_file(file_name);
+  LLVM_DITYPE varSym_type = get_type(varSym->type);
+
+  if(varSym_type){
+    return this->dibuilder.createLocalVariable(
+      llvm::dwarf::DW_TAG_auto_variable, /*Tag*/
+      scope, /* Scope */
+      name, /*Name*/
+      file, /*File*/
+      line_number, /*Lineno*/
+      varSym_type, /*Type*/
+      true/*AlwaysPreserve, won't be removed when optimized*/
+      ); //omit the  Flags and ArgNo
+  }
+  else {
+    //Empty dbg node if the symbol type is unresolved
+    LLVM_DIVARIABLE ret;
+#if HAVE_LLVM_VER >= 37
+    ret = NULL;
+#endif
+    return ret;
+  }
+}
+
+LLVM_DIVARIABLE debug_data::get_variable(VarSymbol *varSym)
+{
+  if( NULL == varSym->llvmDIVariable ){
+    varSym->llvmDIVariable = construct_variable(varSym);
+  }
+  return toDIVARIABLE(varSym->llvmDIVariable);
+}
+
+LLVM_DIVARIABLE debug_data::construct_formal_arg(ArgSymbol *argSym, unsigned ArgNo)
+{
+  const char *name = argSym->name;
+  const char *file_name = argSym->astloc.filename;
+  int line_number = argSym->astloc.lineno;
+  FnSymbol *funcSym = NULL;
+  if(isFnSymbol(argSym->defPoint->parentSymbol))
+    funcSym = (FnSymbol*)argSym->defPoint->parentSymbol;
+  else 
+    printf("Couldn't find the function parent of param: %s!\n",name);
+  
+  LLVM_DISUBPROGRAM scope = get_function(funcSym);  
+  LLVM_DIFILE file = get_file(file_name);
+  LLVM_DITYPE argSym_type = get_type(argSym->type);
+      
+  if(argSym_type)
+    return this->dibuilder.createLocalVariable(
+      llvm::dwarf::DW_TAG_arg_variable, /*Tag: formal arg to a function */
+      scope, /* Scope */
+      name, /*Name*/
+      file, /*File*/
+      line_number, /*Lineno*/
+      argSym_type, /*Type*/
+      true,/*AlwaysPreserve, won't be removed when optimized*/
+      0,  /*Flags*/
+      ArgNo
+      ); 
+  else {
+    //Empty dbg node if the symbol type is unresolved
+    LLVM_DIVARIABLE ret;
+#if HAVE_LLVM_VER >= 37
+    ret = NULL;
+#endif
+    return ret;
+  }
+}
+
+LLVM_DIVARIABLE debug_data::get_formal_arg(ArgSymbol *argSym, unsigned int ArgNo)
+{
+  if( NULL == argSym->llvmDIFormal ){
+    argSym->llvmDIFormal = construct_formal_arg(argSym, ArgNo);
+  }
+  return toDIVARIABLE(argSym->llvmDIFormal);
+}
+/*
+void debug_data::createDeclare(llvm::Value* Storage, LLVM_DIVARIABLE VarInfo)
+{
+  GenInfo* info = gGenInfo; //info is universal across whole project
+
+  if (Storage && VarInfo) 
+  //llvm::Instruction *declareCall = 
+    this->dibuilder.insertDeclare(Storage,VarInfo,info->builder->GetInsertBlock());
+  //declareCall->setDebugLoc();
+  else 
+    printf("CreateDeclare failed !\n");
+}
+*/
```

Comment:
Let's remove this commented out code. It doesn't look like it has any real content.